timelock-server/src/integ2Test/java/com/palantir/atlasdb/timelock/AbstractMultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -262,30 +264,35 @@ public void canCreateNewClientsDynamically() {
     }
 
     @Test
-    @Disabled // TODO(boyoruk): Investigate why this test fails
     public void lockRequestCanBlockForTheFullTimeout() {
-        LockToken token =
+        LockToken token1 =
                 client.lock(LockRequest.of(LOCKS, DEFAULT_LOCK_TIMEOUT_MS)).getToken();
-
+        AtomicReference<LockToken> token2 = new AtomicReference<>(null);
         try {
-            LockResponse response = client.lock(LockRequest.of(LOCKS, LONGER_THAN_READ_TIMEOUT_LOCK_TIMEOUT_MS));
-            assertThat(response.wasSuccessful()).isFalse();
+            Assertions.assertThatThrownBy(() -> {
+                        token2.set(client.lock(LockRequest.of(LOCKS, LONGER_THAN_READ_TIMEOUT_LOCK_TIMEOUT_MS))
+                                .getToken());
+                    })
+                    .hasCause(READ_TIMED_OUT_EXCEPTION);
         } finally {
-            client.unlock(token);
+            client.unlock(token1);
+            if (token2.get() != null) {
+                client.unlock(token2.get());
+            }
```

Comment:
small note: let's expose unlock(Set) so we don't have spurious failures with unlock(token1) failing causing us to leak token 2